packages/flutter/lib/src/widgets/text.dart

```diff
@@ -294,7 +323,7 @@ class Text extends StatelessWidget {
   @override
   void debugFillProperties(DiagnosticPropertiesBuilder description) {
     super.debugFillProperties(description);
-    description.add(new StringProperty('data', data, showName: false));
+    description.add(new StringProperty('data', data ?? span.toPlainText(), showName: false));
```

Comment:
we should include span as its own property, that way it'll appear in the toStringDeep output nicely formatted.

Reply:
Do we generally delegate to the existing debugFillProperty method (like style), or should I write my own PropertyDescription?

Reply:
TextSpan already implements Diagnosticable, so you should be able to do whatever RichText and RenderParagraph do already.

Reply:
Needs work